pex/bin/pex.py

```diff
@@ -290,8 +290,43 @@ def configure_clp_pex_environment(parser):
   group.add_option(
       '--platform',
       dest='platform',
-      default=Platform.current(),
-      help='The platform for which to build the PEX.  Default: %default')
+      default=None,
+      help='The platform for which to build the PEX. Defaults to the platform '
+           'of the running system.')
+
+  group.add_option(
+      '--python-version',
+      dest='python_version',
+      metavar='python_version',
+      default=None,
+      help=("Python version for which to to build the PEX. If not specified, "
+            "then the current system interpreter minor version is used. A major"
+            " version (e.g. '2') can be specified to match all "
+            "minor revs of that major version.  A minor version "
+            "(e.g. '34') can also be specified."),
+  )
+
+  group.add_option(
+      '--implementation',
+      dest='implementation',
+      metavar='implementation',
+      default=None,
+      help=("Python implementation for which to build the PEX (e.g. 'pp', 'jy',"
+            " 'cp',  or 'ip'). If not specified, then the current interpreter "
+            "implementation is used.  Use 'py' to force implementation-"
+            "agnostic wheels."),
```

Comment:
> or you could do combinations of passed in --platform and --abi args like you mentioned. Either seems reasonable to me.

I think the problem with that is that there are no natural combinations here at the ABI level in one of the most common native cases we have/see: a OSX + Linux multi-platform resolve (aka how we build almost every pex at Twitter). OSX interpreters are traditionally always `m` while Linux is traditionally always `mu`. so in that common case, the combinatorial approach would then end up with two completely extraneous platforms (osx->mu, linux->m) which would need to be 1) pre-built/available and 2) resolved - for no apparent benefit. I also think relying on users to pass in position-dependent sets of all 4 facets is likely to be error prone.

so fundamentally to me, it seems like any form of split args for all 4 facets here - which are now necessary to pass all or none - will lead to an awkward, complicated interface for the CLI. having one clear way to pass in the platform + tags triple in all cases would be IMHO cleanest/most ideal - vs having one set of args you pass in for single-platform resolves (e.g. `--platform X --python-version X --implementation X --abi X`) and then an entirely different passing style when you want to pass in a second platform (e.g. `--tags X --tags X`). so I think that means either outright replacing or extending `--platform`.

fwiw, my implicit thinking with the idea of extending `--platform` is largely around backward compatibility. I had alluded briefly to the idea of implicit `any` wildcards for the latter 3 fields, which I think could be a good mechanism for that.

the idea there essentially is that:

```
./pants --platform=linux-x86_64 ...
```

could implicitly map to e.g.:

```
./pants --platform=linux-x86_64-any-any-any ...
```

so that existing users could continue to pass in `--platform` as they do today and get the ~current behavior. then, when the extended form is needed that could be passed in as well just as easily. I was thinking about tackling the backwards compatibility aspect as a fast-follow on top of your work here, prior to cutting a release.

it's also worth noting that this idea of overloading platform and remaining backward compatible would also readily extend to pants as well, where platforms are specified as an iterable of platform strings that could also work with either form to ease the need for migration on users (..tho pants is not exactly affected by the CLI API discussed here, the conceptual mapping and underlying functionality in pex to support is relevant). additionally, `tags` also has an existing conceptual meaning in pants that applies to all targets.

but I definitely hear you re: murky terminology - and this is certainly overloading the term `platform` (and `tags` too, unfortunately).. so I'm open to alternative tacks.

either way tho, it'd probably be a good idea to get some other perspectives on this as well. maybe @jsirois might be able to review and share his take?